src/app/identity.service.ts

```diff
@@ -299,7 +310,11 @@ export class IdentityService {
       encryptedSeedHex,
       this.globalVars.hostname
     );
-    const jwt = this.signingService.signJWT(seedHex);
+    const isDerived =
+      this.accountService.isDerivedKeyAccountFromEncryptedSeedHex(
+        encryptedSeedHex
+      );
+    const jwt = this.signingService.signJWT(seedHex, isDerived);
```

Comment:
I see we're repeating this snippet a decent amount. Can we write a new function on the signingService that handles figuring out if an account is a derived (AKA metamask) account?

Reply:
Declarative is okay IMO